tensorly/decomposition/robust_decomposition.py

```diff
@@ -71,6 +71,8 @@ def robust_pca(X, mask=None, tol=10e-7, reg_E=1, reg_J=1,
     """
     if mask is None:
         mask = 1
+    else:
+        mask = T.tensor(mask, dtype=float)
```

Comment:
MXNet is not properly promoting a boolean/integer-type array when multiplied by a scalar float. The threshold was being converted to 0 below.

Reply:
That was a tricky one, kuddos for finding/fixing it! 
That's a very odd behavior from MXNet -- we could report in the main repo.

Reply:
Definitely was tricky—I got lucky by seeing `rec_X` was always 0 with a mask.

https://github.com/apache/incubator-mxnet/issues/19891